src/mouse.rs

```diff
@@ -162,10 +177,30 @@ impl From<Button> for CGMouseButton {
 #[cfg(target_os = "macos")]
 fn system_move_to(point: Point) {
     let point = CGPoint::from(point);
-    let source = CGEventSource::new(HIDSystemState).unwrap();
-    let event =
-        CGEvent::new_mouse_event(source, CGEventType::MouseMoved, point, CGMouseButton::Left);
-    event.unwrap().post(CGEventTapLocation::HID);
+
+    macro_rules! send {
+        ($event_type:expr, $btn:expr) => {
+            let source = CGEventSource::new(HIDSystemState).unwrap();
+            let event = CGEvent::new_mouse_event(source, $event_type, point.clone(), $btn);
+            event.unwrap().post(CGEventTapLocation::HID);
+        };
+    }
+
+    let button_bits: u16 = *mac_down_buttons.read().unwrap();
+
+    if button_bits & u16::from(Button::Left) != 0 {
+        send!(CGEventType::LeftMouseDragged, CGMouseButton::Left);
+    }
+    if button_bits & u16::from(Button::Right) != 0 {
+        send!(CGEventType::RightMouseDragged, CGMouseButton::Right);
+    }
+    if button_bits & u16::from(Button::Middle) != 0 {
+        send!(CGEventType::OtherMouseDragged, CGMouseButton::Center);
+    }
+    if button_bits == 0 {
+        send!(CGEventType::MouseMoved, CGMouseButton::Left);
+    }
+    
```

Comment:
I see why all of this makes sense. I did try your changes with the new tests though and strangely drag and drop test still failed (they do fail at the time being which is why we skip them but I was hoping this will make them pass so we can validate this works for all time). Will it be ok for you to rebase on the most recent main branch and retry testing this manually with Rust Edition 2024?